File: VB/T115891VB/Controllers/HomeController.vb
Imports DevExpress.Web.Mvc
Public Class HomeController
    Inherits System.Web.Mvc.Controller

    Function Index() As ActionResult
        Return View()
    End Function

    <ValidateInput(False)>
    Public Function GridViewPartial() As ActionResult
        Return PartialView("_GridViewPartial", BatchEditRepository.GridData)
    End Function

    <HttpPost, ValidateInput(False)>
    Public Function BatchUpdatePartial(ByVal batchValues As MVCxGridViewBatchUpdateValues(Of GridDataItem, Integer)) As ActionResult
        For Each item In batchValues.Insert
            If batchValues.IsValid(item) Then
                BatchEditRepository.InsertNewItem(item, batchValues)
            Else
                batchValues.SetErrorText(item, "Correct validation errors")
            End If
        Next item
        For Each item In batchValues.Update
            If batchValues.IsValid(item) Then
                BatchEditRepository.UpdateItem(item, batchValues)
            Else
                batchValues.SetErrorText(item, "Correct validation errors")
            End If
        Next item
        For Each itemKey In batchValues.DeleteKeys
            BatchEditRepository.DeleteItem(itemKey, batchValues)
        Next itemKey
        Return PartialView("_GridViewPartial", BatchEditRepository.GridData)
    End Function
    Public Function GridViewCustomActionPartial(ByVal key As String) As ActionResult
        Session("Mode") = key
        Return PartialView("_GridViewPartial", BatchEditRepository.GridData)
    End Function
End Class

File: CS/T115891/Controllers/HomeController.cs
using DevExpress.Web.Mvc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using T115891.Models;

namespace T115891.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
            return View();
        }

        [ValidateInput(false)]
        public ActionResult GridViewPartial()
        {
            return PartialView("_GridViewPartial", BatchEditRepository.GridData);
        }

        [HttpPost, ValidateInput(false)]
        public ActionResult BatchUpdatePartial(MVCxGridViewBatchUpdateValues<GridDataItem, int> batchValues)
        {
            foreach (var item in batchValues.Insert)
            {
                if (batchValues.IsValid(item))
                    BatchEditRepository.InsertNewItem(item, batchValues);
                else
                    batchValues.SetErrorText(item, "Correct validation errors");
            }
            foreach (var item in batchValues.Update)
            {
                if (batchValues.IsValid(item))
                    BatchEditRepository.UpdateItem(item, batchValues);
                else
                    batchValues.SetErrorText(item, "Correct validation errors");
            }
            foreach (var itemKey in batchValues.DeleteKeys)
            {
                BatchEditRepository.DeleteItem(itemKey, batchValues);
            }
            return PartialView("_GridViewPartial", BatchEditRepository.GridData);
        }
        public ActionResult GridViewCustomActionPartial(string key)
        {
            Session["Mode"] = key;
            return PartialView("_GridViewPartial", BatchEditRepository.GridData);
        }
    }
}